flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobUtilsTest.java

```diff
@@ -71,4 +77,44 @@ public void testTaskManagerFallbackBlobStorageDirectory1() throws IOException {
     public void testBlobUtilsFailIfNoStorageDirectoryIsSpecified() throws IOException {
         BlobUtils.createBlobStorageDirectory(new Configuration(), null);
     }
+
+    @Test
+    public void testCheckAndDeleteCorruptedBlobsDeletesCorruptedBlobs() throws IOException {
+        final File storageDir = temporaryFolder.newFolder();
+
+        final BlobKey corruptedBlobKey = BlobKey.createKey(BlobKey.BlobType.PERMANENT_BLOB);
+        final byte[] corruptedContent = "corrupted".getBytes(StandardCharsets.UTF_8);
+
+        final byte[] validContent = "valid".getBytes(StandardCharsets.UTF_8);
+        final byte[] validKey = BlobUtils.createMessageDigest().digest(validContent);
+        final BlobKey validPermanentBlobKey =
+                BlobKey.createKey(BlobKey.BlobType.PERMANENT_BLOB, validKey);
+        final BlobKey validTransientBlobKey =
+                BlobKey.createKey(BlobKey.BlobType.TRANSIENT_BLOB, validKey);
+
+        writeBlob(storageDir, corruptedBlobKey, corruptedContent);
+        writeBlob(storageDir, validPermanentBlobKey, validContent);
+        writeBlob(storageDir, validTransientBlobKey, validContent);
+
+        BlobUtils.checkAndDeleteCorruptedBlobs(storageDir.toPath(), log);
+
+        assertThat(
+                BlobUtils.listBlobsInDirectory(storageDir.toPath()).stream()
+                        .map(BlobUtils.Blob::getBlobKey)
+                        .collect(Collectors.toList()),
+                containsInAnyOrder(validPermanentBlobKey, validTransientBlobKey));
+    }
+
+    private void writeBlob(File storageDir, BlobKey corruptedBlobKey, byte[] fileContent)
```

Comment:
rename `corruptedBlobKey`

Reply:
I think this problem has been solved with 63201b3abe1c4c55383ca34130c6c49483c683ed.